MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2014,19 +1996,28 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
             }
 
             // If we're hardware VAAPI decoding and software encoding, download frames from the decoder first
-            else if (string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase)
+            else if (IsVaapiSupported(state) && string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase)
                 && string.Equals(outputVideoCodec, "libx264", StringComparison.OrdinalIgnoreCase))
             {
                 var codec = videoStream.Codec.ToLowerInvariant();
-                var pixelFormat = videoStream.PixelFormat.ToLowerInvariant();
-
-                // Assert hardware VAAPI decodable (Except h264 10-bit and higher color depth)
-				// TODO: a propery way to detect hardware capabilities and falling back when transcoding is failed
-                if ((pixelFormat ?? string.Empty).IndexOf("p10", StringComparison.OrdinalIgnoreCase) == -1
-                    || ((pixelFormat ?? string.Empty).IndexOf("p10", StringComparison.OrdinalIgnoreCase) != -1
-                        && (string.Equals(codec, "hevc", StringComparison.OrdinalIgnoreCase)
-                            || string.Equals(codec, "h265", StringComparison.OrdinalIgnoreCase)
-                            || string.Equals(codec, "vp9", StringComparison.OrdinalIgnoreCase))))
+                var IsColorDepth10 = (videoStream.Profile ?? string.Empty).IndexOf("10", StringComparison.OrdinalIgnoreCase) != -1;
```

Comment:
this feels somewhat wrong... cannot we look for a wider substring to be safe?

Reply:
One problem here is that `videoStream.BitDepth` is only valid for H264 video, but not for HEVC and MPEG4 video (maybe the same for other formats).

I am not sure when this parameter was detected. Maybe it is when scanning in the media library? I'm not too familiar with that.

Reply:
Neither is me. I'm just on the wary side of assuming a `10` in profile would always mean "10-bit"...

Reply:
`High 10` or `Main 10` is the complete substring. But beyond this PR, I think it is necessary to make `videoStream.BitDepth` available for any video.